snooty/postprocess.py

```diff
@@ -546,6 +547,80 @@ def __call__(self, fileid_stack: FileIdStack, node: n.Node) -> None:
             )
 
 
+class BannerHandler:
+    """Traverse a series of pages matching specified targets in Snooty.toml
+    and append Banner directive nodes"""
+
+    def __init__(self, banners: List[BannerInstantiated], root: Path) -> None:
+        self.banners = banners
+        self.root = root
+
+    def __find_target_insertion_node(self, node: n.Parent[n.Node]) -> Optional[n.Node]:
+        """Search via BFS for the first 'section' from a root node, arbitrarily terminating early if
+        no 'section' is found within the first 50 nodes."""
+        queue: List[n.Node] = list(node.children)
+        curr_iteration = 0
+        max_iteration = 50
+
+        insertion_node = None
+
+        while queue and curr_iteration < max_iteration:
+            candidate = queue.pop(0)
+            if candidate.type == "section":
+                insertion_node = candidate
+                break
+            if isinstance(candidate, n.Parent):
+                queue.extend(candidate.children)
+
+            curr_iteration += 1
+        return insertion_node
+
+    def __determine_banner_index(self, node: n.Parent[n.Node]) -> int:
+        """Determine if there's a heading within the first level of the target insertion node's children.
+        If so, return the index position after the first detected heading. Otherwise, return 0."""
+        return (
+            next(
+                (
+                    idx
+                    for idx, child in enumerate(node.children)
+                    if isinstance(child, n.Heading)
+                ),
+                0,
+            )
+            + 1
+        )
+
+    def __page_target_match(
+        self, targets: List[str], page: Page, fileid: FileId
+    ) -> bool:
+        """Check if page matches target specified, but always return false for includes"""
+        if fileid.suffix != ".txt":
```

Comment:
Not sure this branch will ever be taken: it's only called from enter_page(), which should only be called on files that are already terminated with `.txt` based on logic in `Postprocessor.run_event_parser()`.

Might make sense just to make this an assert to make sure?